src/playercards/customizable/UpgradeSheetLibrary.ttslua

```diff
@@ -0,0 +1,364 @@
+-- static values
+Y_VISIBLE    = 0.25
+Y_INVISIBLE  = -0.5
+
+-- used for Summoned Servitor and Living Ink
+VECTOR_COLOR = {
+  unselected = { 0.5, 0.5, 0.5, 0.75 },
+  selected   = { 0.597, 0.195, 0.796 }
+}
+
+-- used for Summoned Servitor
+-- these match with ArkhamDB's way of storing the data in the dropdown menu
+SLOT_INDICES = { arcane = "1", ally = "0", none = "" }
+
+-- override 'marked boxes' for debugging ('all' or 'none')
+DEBUG        = ""
+
+-- save state when going into bags / decks
+function onDestroy() self.script_state = onSave() end
+
+function onSave()
+  -- Living Ink custom save function
+  if selfId == "09079-c" then
+    inputValues = {}
+    for skill, isSelected in pairs(selectedSkills) do
+      if isSelected then
+        table.insert(inputValues, skill)
+      end
+    end
+  end
+  return JSON.encode({ markedBoxes, inputValues })
+end
+
+-- Startup procedure
+function onLoad(savedData)
+  if savedData ~= "" and DEBUG == "" then
+    local loadedData = JSON.decode(savedData)
+    markedBoxes      = loadedData[1]
```

Comment:
These really shouldn't be separate arrays.  It should be a single array of tables:
{
  xp = 0,
  text = ""
}

This keeps the boxes and field together so there's no need to correlate them, much less correlate them across different indexes which is very confusing.

I'd suggest a different name here as well, something more to what the data represents